src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs

```diff
@@ -534,6 +534,7 @@ private void SetGenericValueImpl<T>(int pos, ref T value)
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         private extern void SetValueRelaxedImpl(object? value, int pos);
 
+#pragma warning disable CA1822
```

Comment:
Why are these different? Is it just the volume and number of callers?

Reply:
I think the methods which are accessing `Length` property can be excluded from this suppression?

Reply:
They are accessed as instance calls from JIT

Reply:
Looks like `InternalArray__ICollection_get_Count`, `InternalArray__IReadOnlyCollection_get_Count` and few others which are accessing class members do not need the suppression.

Reply:
The intention is to indicate that none of the methods should be made static even if they could